autogen/oai/gemini.py

```diff
@@ -0,0 +1,303 @@
+"""Create a OpenAI-compatible client for Gemini features.
+
+Resources:
+- https://ai.google.dev/docs
+- https://cloud.google.com/vertex-ai/docs/generative-ai/migrate-from-azure
+- https://blog.google/technology/ai/google-gemini-pro-imagen-duet-ai-update/
+- https://ai.google.dev/api/python/google/generativeai/ChatSession
+"""
+
+from __future__ import annotations
+
+import base64
+import os
+import random
+import re
+import time
+from io import BytesIO
+from typing import Any, Dict, List, Mapping, Union
+
+import google.generativeai as genai
+import requests
+from google.ai.generativelanguage import Content, Part
+from google.api_core.exceptions import InternalServerError
+from openai.types.chat import ChatCompletion
+from openai.types.chat.chat_completion import ChatCompletionMessage, Choice
+from openai.types.completion_usage import CompletionUsage
+from PIL import Image
+
+# from autogen.agentchat.contrib.img_utils import _to_pil, get_image_data
+from autogen.token_count_utils import count_token
+
+
+class GeminiClient:
+    """Client for Google's Gemini API.
+
+    TODO: this Gemini implementation does not support the following features yet:
+    - function_call (OpenAI)
+    - tool_calls (OpenAI)
+    - safety_setting (Gemini)
+    - Multi-turn chat for vision model (Gemini)
+    - multiple responses at the same time (Gemini)
+    """
+
+    def __init__(self, **kwargs):
+        self.api_key = kwargs.get("api_key", None)
+        if self.api_key is None:
+            self.api_key = os.getenv("GOOGLE_API_KEY")
+
+        assert self.api_key is not None, (
+            "Please provide api_key in OAI_CONFIG_LIST " "or set the GOOGLE_API_KEY env variable."
```

Comment:
I think instead of mentioning only OAI_CONFIG_LIST, we should say "Please provide api_key in your config list entry for Gemini".